diffusion/datasets/image_caption_latents.py

```diff
@@ -140,6 +143,15 @@ def __getitem__(self, index):
                 if 'CLIP_LATENTS' in latent_key:
                     clip_pooled = np.frombuffer(sample[f'{caption_key}_CLIP_POOLED_TEXT'], dtype=np.float32).copy()
                     out['CLIP_POOLED'] = torch.from_numpy(clip_pooled).to(self.latent_dtype).reshape(latent_shape[1])
+        if self.drop_nans:
+            if out['CLIP_POOLED'].isnan().any():
```

Comment:
this will work for us, but this section will fail if the text latent keys are named differently - probably safer to loop thru `self.text_latent_keys` and `self.attention_mask_keys`, e.g. something like:

``` 
if self.drop_nans:
    for latent_key, attn_key in zip(self.text_latent_keys, self.attention_mask_keys):
        if out[latent_key].isnan().any():
            out[latent_key] = torch.zeros_like(out[latent_key])
            out[attn_key] = torch.zeros_like(out[attn_key])
        if 'CLIP_LATENTS' in latent_key:
            out['CLIP_POOLED'] = torch.zeros_like(out['CLIP_POOLED'])
```

Reply:
This should be fixed now!